src/app/storage/duck/operations.ts

```diff
@@ -67,12 +110,49 @@ const fetchStorage = () => {
       );
       const res = await client.list(resource);
       //temporary for ui work
-      dispatch(migStorageFetchSuccess(res.data.items));
+      const migStorages = res.data.items;
+      const refs = await Promise.all(
+        fetchMigStorageRefs(client, migMeta, migStorages),
+      );
+      const groupedStorages = groupStorages(migStorages, refs);
+      dispatch(migStorageFetchSuccess(groupedStorages));
     } catch (err) {
       dispatch(AlertCreators.alertError(err));
     }
   };
 };
+function fetchMigStorageRefs(
+  client: IClusterClient,
+  migMeta,
+  migStorages,
+): Array<Promise<any>> {
+  const refs: Array<Promise<any>> = [];
+
+  migStorages.forEach(storage => {
+    const secretRef = storage.spec.backupStorageConfig.credsSecretRef;
+    const secretResource = new CoreNamespacedResource(
+      CoreNamespacedResourceKind.Secret,
+      secretRef.namespace,
+    );
+    refs.push(client.get(secretResource, secretRef.name));
+  });
+
+  return refs;
+}
+
+function groupStorages(migStorages: any[], refs: any[]): any[] {
```

Comment:
We shoud chat about how to best represent these objects in the front end. I'm curious your opinions on that but you nailed what I did for clusters.

The general opinion I took was, our front end consumes a "cluster". In reality, thats 3 objects. So in the redux tree,  you can get a `cluster.clusterList.{Cluster,MigCluster,Secret}` to retrieve the individual data.